view_conversations/conv-viewer-webapp/src/app/app.component.ts

```diff
@@ -94,17 +122,65 @@ export class AppComponent implements OnInit {
     document.location.hash = JSON.stringify(objToEncode);
   }
 
+  submitCommentSearch(comment : wpconvlib.Comment) {
+    console.log('model-based form submitted');
+    this.errorMessage = null;
+    this.updateLocationHash();
+
+    this.inFlightRequest =
+        this.http
+            .get(encodeURI(
+              '/api/comment-id/' + comment.id))
+            .subscribe(
+                (actions: wpconvlib.Comment[]) => {
+                  console.log('got conversation!');
+                  this.searchResult = JSON.stringify(actions, null, 2);
+                  delete this.inFlightRequest;
+
+                  const conversation: wpconvlib.Conversation = {};
+                  for (const a of actions) {
```

Comment:
This little for loop seems to be doing some very self-contained logic about comment highlighting. Maybe make it it own simple function that you can unit test the behaviour of?